exporter/datadogexporter/factory.go

```diff
@@ -514,11 +535,18 @@ func (f *factory) createLogsExporter(
 			}
 			return nil
 		}
-	} else {
+	case isLogsAgentExporterEnabled():
+		logsAgent, exp, err := newLogsAgentExporter(ctx, set, cfg, hostProvider)
+		if err != nil {
+			cancel()
+			return nil, err
+		}
+		f.logsAgent = logsAgent
+		pusher = exp.ConsumeLogs
+	default:
 		exp, err := newLogsExporter(ctx, set, cfg, &f.onceMetadata, attributesTranslator, hostProvider, metadataReporter)
 		if err != nil {
 			cancel()
-			f.wg.Wait() // then wait for shutdown
```

Comment:
Is this change unrelated to the PR?

Reply:
Somewhat unrelated, but I was looking into this line and saw that `f.wg` is only used in the `consumeStatsPayload` and `TraceAgent` methods, which are never used in `createLogsExporter`. So I don't think this call was required here

Reply:
Agree is not required, but I think we can do any cleanup in a different PR. This PR is already big with only the needed changes :smile: